### PR TITLE
storage: add `test_block_body_indices` for `block_body_indices`

### PR DESCRIPTION
**Pull request by tcoratger**
Related https://github.com/paradigmxyz/reth/issues/10338

### CODE REVIEW

**Comment by shekhirin** [comment]
Blocked by https://github.com/paradigmxyz/reth/pull/10356, want to get it in first because it'll cause a bunch of conflicts

**Comment by emhane** [comment]
unblocked @shekhirin

**Comment by tcoratger** [comment]
> unblocked @shekhirin
@emhane Just rebased and fixed conflicts

File: crates/storage/provider/src/providers/blockchain_provider.rs [hunk 1]
```diff
@@ -1861,6 +1865,87 @@ mod tests {
...
+        let chain = NewCanonicalChain::Commit {
+            new: vec![ExecutedBlock::new(
+                Arc::new(first_in_mem_block),
+                Arc::new(in_memory_block_senders),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+            )],
+        };
+        provider.canonical_in_memory_state.update_chain(chain);
```
Comment (shekhirin) [inline crates/storage/provider/src/providers/blockchain_provider.rs:1923]: can we use `provider_with_random_blocks` here?

**Review by shekhirin — DISMISSED** [review]
Can use the new helper, otherwise LGTM.

File: crates/storage/provider/src/providers/blockchain_provider.rs (hunk 1 above)
Comment (tcoratger) [reply crates/storage/provider/src/providers/blockchain_provider.rs:1923]: Here I don't think it's a good idea. Indeed, to properly test the results, I inserted random transactions in the in memory and in database blocks, which is not done in the `provider_with_random_blocks` function. 

When I try to add it to the function directly, it breaks other tests and it's probably not useful elsewhere to add this additional complexity to the util. So I would say that we could stay on this custom made initialization for the test here and that we could eventually see once the tests are all implemented for a more custom and reliable utils test suite, what do you think?

Comment (shekhirin) [reply crates/storage/provider/src/providers/blockchain_provider.rs]: makes sense! thanks for the explanation

**Review by mattsse — APPROVED** [review]

**Pull request merged**